lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -7206,10 +7200,7 @@ GenericSignature GenericSignatureBuilder::computeGenericSignature(
   // will produce the same thing.
   //
   // We cannot do this when there were errors.
-  // FIXME: The HadAnyRedundantConstraints bit is a hack because we are
-  // over-minimizing.
-  if (allowBuilderToMove && !Impl->HadAnyError &&
```

Comment:
I bet that removing this hack reduces the number of GSB's we create when (e.g.) building the standard library.

Reply:
I saw stdlib build time with a release-asserts compiler decrease from 88.05 seconds to 86.50 seconds (this is after building it twice, to skip the LLVM code generation; that adds another 60 seconds or so).